salt/dex/init.sls

```diff
@@ -0,0 +1,124 @@
+include:
+  - crypto
+  - repositories
+  - kubectl-config
+
+{% set ip_addresses = [] -%}
+{% set extra_names = ["DNS: " + grains['caasp_fqdn'] ] -%}
+{% if "kube-master" in salt['grains.get']('roles', []) %}
+  {% do ip_addresses.append("IP: " + pillar['api']['cluster_ip']) %}
+  {% for _, interface_addresses in grains['ip4_interfaces'].items() %}
+    {% for interface_address in interface_addresses %}
+      {% do ip_addresses.append("IP: " + interface_address) %}
+    {% endfor %}
+  {% endfor %}
+  {% for extra_ip in pillar['api']['server']['extra_ips'] %}
+    {% do ip_addresses.append("IP: " + extra_ip) %}
+  {% endfor %}
+  # add some extra names the API server could have
+  {% set extra_names = extra_names + ["DNS: dex",
+                                      "DNS: dex.kube-system",
+                                      "DNS: dex.kube-system.svc",
+                                      "DNS: dex.kube-system.svc." + pillar['internal_infra_domain'],
+                                      "DNS: api." + pillar['internal_infra_domain']] %}
+  {% for extra_name in pillar['api']['server']['extra_names'] %}
+    {% do extra_names.append("DNS: " + extra_name) %}
+  {% endfor %}
+
+  # add the fqdn provided by the user
+  # this will be the name used by the kubeconfig generated file
+  {% if salt['pillar.get']('api:server:external_fqdn') %}
+    {% do extra_names.append("DNS: " + pillar['api']['server']['external_fqdn']) %}
+  {% endif %}
+{% endif %}
```

Comment:
We probably want to move this whole block to a jinja macro... Similar code (if not the same) exists in https://github.com/kubic-project/salt/blob/1e41512047a193d30ad06f78e5fe7a19faeb4824/salt/cert/init.sls#L4-L32 and https://github.com/kubic-project/salt/blob/930ea6e8ee03555c27b018f0af93bb65a03e423d/salt/kube-apiserver/init.sls#L8-L40

Reply:
Fair point - they're similar but I think may have slightly different names they generate.